waspc/data/Generator/templates/server/src/auth/providers/oauth/env.ts

```diff
@@ -0,0 +1,21 @@
+import { type ProviderConfig } from "wasp/auth/providers/types";
+
+export function ensureEnvVarsForProvider<EnvVars extends string>(
+  envVarNames: EnvVars[],
+  provider: ProviderConfig,
+): {
+  [name in EnvVars]: string;
+} {
+  const result: {
+    [name: string]: string;
+  } = {};
```

Comment:
I see that sometimes you use index signatures, while other times you use Records.

If you're going to mix and match, I recommend:
- Records for bounded types and/or when you want to document the key's name
- Index signatures for unbounded types and when you don't care about documenting the key's name

See here for details: https://github.com/wasp-lang/wasp/pull/1235#discussion_r1300022520